hrjobcontract/js/src/job-contract/controllers/modal/modal-contract.js

```diff
@@ -53,6 +54,9 @@ define([
             angular.copy(entity,$scope.entity);
             angular.copy(files,$scope.files);
 
+            $scope.entity.details.period_start_date = convertToDateObject($scope.entity.details.period_start_date);
```

Comment:
Probably changes to 'filterDate' filter were introduced later, but it may be worth changing.
```javascript
$scope.entity.details.period_start_date = $filter('formatDate')($scope.entity.details.period_start_date, Date);
```
Or is there any other reason you don't use this filter here?